server/src/main/java/io/deephaven/server/session/SessionState.java

```diff
@@ -648,7 +651,9 @@ private synchronized void setDependencies(final List<ExportObject<?>> parents) {
 
             this.parents = parents;
             dependentCount = parents.size();
-            parents.stream().filter(Objects::nonNull).forEach(this::tryManage);
+            alreadyDeadParent = parents.stream()
+                    .filter(p -> p != null && !tryManage(p))
```

Comment:
tryManage is moving to an intermediate operation here as opposed to a terminal operation; this has the potential to not execute tryManage for certain parents.

Reply:
I think this is acceptable/desirable under the circumstances. See https://github.com/deephaven/deephaven-core/pull/5014#discussion_r1445515975

Reply:
I disagree. Trying to (ab)use stream with stateful intermediate ops is poor practice even if we think behavior seems ok in this case. Much safer and cleaner to just use for loop and break out for explicitly defined flow control.

Reply:
No biggie. I've pulled this into a traditional for loop.